src/main/java/io/vertx/httpproxy/spi/cache/Cache.java

```diff
@@ -1,9 +1,47 @@
 package io.vertx.httpproxy.spi.cache;
 
-import java.util.Map;
+import io.vertx.core.Future;
+import io.vertx.core.dns.SrvRecord;
+
 
 /**
  * Cache SPI.
  */
-public interface Cache<K, V> extends Map<K, V> {
+public interface Cache {
+
+  /**
+   * Being called when the proxy attempts to add a new cache item.
+   * The cache can only store up to maxSize of the latest items based
+   * on CacheOptions.
+   *
+   * @param key the URI of the resource
+   * @param value the cached response
+   * @return a succeed void future
+   */
+  Future<Void> put(String key, Resource value);
+
+  /**
+   * Being called when the proxy attempts to fetch a cache item.
+   *
+   * @param key the URI of the resource
+   * @return the cached response, null if not exist
```

Comment:
I think we should return a Future with null result instead of null. Some cache implementation might actually check the disk

Reply:
Got it, this is a document issue, I didn't state it well